lightly/transforms/random_crop_and_flip_with_grid.py

```diff
@@ -79,22 +64,51 @@ class RandomHorizontalFlipReturnsIfFlip(T.RandomHorizontalFlip):
         p (float): probability of the image being flipped. Default value is 0.5
     """
 
-    def forward(self, img: ImageAndLocation) -> ImageAndLocation:
+    def forward(self, img: Image.Image, location:Location) -> Tuple[Image.Image, Location]:
         """
         Args:
-            img (ImageAndLocation): ImageAndLocation to be flipped.
+            img (PIL Image or Tensor): Image to be flipped..
+            Location: Location object linked to the image
         Returns:
-            ImageAndLocation: Randomly flipped image with updated location.flip parameter
-
+            PIL Image or Tensor: Randomly flipped image 
+            Location: Location object with updated location.horizontal_flip parameter
         """
         
         if torch.rand(1) < self.p:
-            img.image = F.hflip(img.image)
-            img.location.flip = True
-        return img
+            img = F.hflip(img)
+            location.horizontal_flip = True
```

Comment:
Technically this should be `location.horizontal_flip = not location.horizontal_flip` because you could have multiple flip transformations. But its fine if you leave it like that.